### PR TITLE
[REVIEW] Raise error for list like dtypes in cudf

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
Raising error `cudf Column doesn't support list like data types` for list like elements.

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -1240,6 +1240,11 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
                 size=pa_size,
                 offset=pa_offset,
             )
+        elif isinstance(arbitrary, pa.ListArray):
+            raise ValueError(
+                "cudf Column doesn't support list like data types"
+            )
```
Comment (shwina) [inline python/cudf/cudf/core/column/column.py]: Let's use `NotImplementedError` as this is something we don't yet support, as opposed to a problem with user-provided data. Also let's avoid using the term `Column` in public errors. 

```suggestion
            raise NotImplementedErr(
                "cudf doesn't support list like data types"
            )
```

File: python/cudf/cudf/tests/test_dtypes.py [hunk 2]
```diff
@@ -34,3 +34,35 @@ def test_cdf_to_pandas(data, ordered):
         pd.CategoricalDtype(data, ordered)
         == cudf.CategoricalDtype(categories=data, ordered=ordered).to_pandas()
     )
+
+
+@pytest.mark.parametrize(
+    "data", [{"a": [[1, 2, 3]]}, pd.DataFrame({"a": [[1, 2, 3]]})]
```
Comment (shwina) [inline python/cudf/cudf/tests/test_dtypes.py]: Could we add more tests here to check that we raise the same error for the below cases for example?

* Empty list?
* More than one list
* Combination of list and numeric data?

Ditto below in DataFrame.

**Comment by rgsl888prabhu** [comment]
@shwina @kkraus14

**Comment by ntaherkhani** [comment]
Hi, 
I want to have adask-cudf  included couple string and int columns and one column as list( python list or ndarray). I am getting " cudf Column doesn't support list like data types" . what should define a df including a column with list data type?

**Comment by rgsl888prabhu** [comment]
> Hi,
> I want to have adask-cudf included couple string and int columns and one column as list( python list or ndarray). I am getting " cudf Column doesn't support list like data types" . what should define a df including a column with list data type?
> 
> thanks

Which cudf version are you using, as of cudf 0.15 should be supporting list.

**Comment by ntaherkhani** [comment]
thanks for your quick response.
I am using rapidsai  and my cudf version is 0.14.0 
cudf                     0.14.0
cugraph                  0.14.0+0.g9efe553.dirty
cuml                     0.14.0+0.g31202da.dirty
cupy                     7.6.0
cusignal                 0+unknown
cuspatial                0.14.0
cuxfilter                0.14.0
cycler                   0.10.0
cymem                    2.0.3
Cython                   0.29.17
cytoolz                  0.10.1
dask                     2.20.0
dask-cuda                0.14.0
dask-cudf                0.14.0
dask-glm                 0.2.0
dask-labextension        2.0.2
dask-ml                  1.5.0

as you mentioned cudf 15.0 is supporting list data type for column I am going to upgrade my cudf . Should I upgrade all of the other libs in the list?

**Comment by rgsl888prabhu** [comment]
If you are using conda, it should take care if I am not wrong.

**Pull request merged**